src/actions/app.js

```diff
@@ -28,10 +30,31 @@ export const toggleEditState = () => ({
   type: TOGGLE_EDITING_STATE,
 });
 
-export const saveAnnotation = (annotationData) => ({
-  type: SAVE_ANNOTATION,
-  payload: annotationData,
-});
+export const saveAnnotation = (annotationData, id = null) => {
+  return (dispatch, getState) => {
+    dispatch({
+      type: SAVE_ANNOTATION,
+      payload: annotationData,
+    });
+    // If the snippet has not been saved, then saving an annotation shouldn't
+    // trigger a POST request to save the snippet (and its annotations)
+    if (!id) {
+      // Return a Promise object to make this action "thenable"
+      return Promise.resolve();
+    }
+    // Get the app state and stringify it
+    const stateString = JSON.stringify(getState().app);
+
+    // TODO: Dispatch action to show the snackbar with a relevant message based on the results
+    // Return the results of the POST request
+    return axios.post(`api/snippets/${id}`, { json: stateString, })
```

Comment:
> Also, we need the state after the SAVE_ANNOTATION action. In the Annotations container, we only have the state before that action. We could merge the two together in the handler function, but that's the job of the reducer, not the container. By using thunk, we're able to call getState() after we've dispatched the SAVE_ANNOTATION action, so that we can easily save the state to Redis.